File: Domain/Todo.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Domain
{
    public enum State
    {
        DONE,
        IN_PROGRESS,
        DROPPED
    }
    public class Todo
    {
        public string Id { get; set; } = Guid.NewGuid().ToString();
        public required string Title { get; set; }
        public required string Description { get; set; }
        public TimeSpan EstimatedTime { get; set; }
        public TimeSpan TimeLogged { get; set; }
        public State State { get; set; }
        public required string UserId { get; set; }
        public User User { get; set; } = null!;
    }
}

File: client/src/lib/types/todo.ts
export const StateVal = {
    DONE: 0,
    IN_PROGRESS: 1,
    DROPPED: 2
} as const;

export type State = typeof StateVal[keyof typeof StateVal];

export interface Todo {
    id: string;
    title: string;
    description: string;
    estimatedTime?: string; // Serialized C# TimeSpan ("hh:mm:ss")
    timeLogged?: string;    // Serialized C# TimeSpan ("hh:mm:ss")
    state: State;
    userId: string;
}